%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.                                                 %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.1 Generated 2015/22/05 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%

\documentclass{frontiersSCNS} % for Science, Engineering and Humanities and Social Sciences articles
%\documentclass{frontiersHLTH} % for Health articles
%\documentclass{frontiersFPHY} % for Physics and Applied Mathematics and Statistics articles

%\setcitestyle{square}
\usepackage{url,hyperref,lineno,microtype}
\usepackage[onehalfspacing]{setspace}
\usepackage{subcaption}

\linenumbers


% Leave a blank line between paragraphs instead of using \\


\def\keyFont{\fontsize{8}{11}\helveticabold }
\def\firstAuthorLast{Camargo {et~al.}} %use et al only if is more than 1 author
\def\Authors{Anyela V Camargo\,$^{1,*}$, Jan T Kim\,$^{{2}}$, Alison Bentley\,$^{{3,*}}$}
% Affiliations should be keyed to the author's name with superscript numbers and be listed as follows: Laboratory, Institute, Department, Organization, City, State abbreviation (USA, Canada, Australia), and Country (without detailed address information such as city zip codes or street names).
% If one of the authors has a change of address, list the new address below the correspondence details using a superscript symbol and use the same symbol to indicate the author in the author list.
\def\Address{$^{1}$National Plant Phenomics Centre, IBERS, Aberyswyth University, Gogerddan, Aberystwyth, SY23 3EB, UK\\
$^{3}$ The John Bingham Laboratory, NIAB, UK\\ $^{2}$ The Pirbright Institute, UK}
% The Corresponding Author should be marked with an asterisk
% Provide the exact contact address (this time including street name and city zip code) and email of the corresponding author
\def\corrAuthor{Alison Bentley}
\def\corrAddress{The John Bingham Laboratory, NIAB, UK}
\def\corrEmail{alison.bentley@niab.com}

\newcommand{\todo}[1]{
  \rule{0pt}{0pt}\marginpar{{\color{blue}\rule{1ex}{1ex}}}
  {[\textbf{\color{blue}todo:} #1]}}


\begin{document}
\onecolumn
\firstpage{1}

\title[Running Title]{Article Title} 

\author[\firstAuthorLast ]{\Authors} %This field will be automatically populated
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}


\maketitle



\begin{abstract}

\tiny
 \keyFont{ \section{Keywords:} wheat senescence diseases datascience phenomics } %All article types: you may provide up to 8 keywords; at least 5 are mandatory.
\end{abstract}

\section{Introduction}

Senescence is the last developmental stage of plant cells, tissues, organs, and, in the case of monocarpic species, the entire plant. Once senescence has been initiated, it typically leads to a massive remobilization of phloem-mobile nutrients from the senescing plant parts to developing sinks, such as seeds and grains of monocarpic crops \cite{Distelfeld27012014}. Plant senescence can be triggered by a range of abiotic and biotic factors such as light, UV, ozone, hormones, drought and various pests \cite{PLB:PLB114} Fig \ref{fig1:flowchart}. Pathogen infections interact with developmental processes in a complex way. The outcome of this interaction may greatly influence plant productivity as well as resistance and susceptibility. Pathogenesis and development mutually influence each other. On the one hand, developmental conditions of the host plant may determine the outcome of pathogen infection. On the other hand, pathogen infection can change the developmental program of the host. Symptoms of senescence often accompany the progression of disease. In other cases, senescence is delayed in response to pathogen infection. The lifestyle of the pathogen plays an important role for the developmental response of the host. Necrotrophic pathogens derive nutrients from dead cells, whereas biotrophic pathogens need living cells to meet their nutritional demands. Between both extremes, a third nutritional type is recognized: the hemibiotrophs. Hemibiotrophs adopt both lifestyles, switching from a biotrophic phase at the beginning of the infection to a necrotrophic lifestyle as pathogenesis proceeds. Generally, senescence and cell death are regarded as favorable to the necrotrophic lifestyle, and necrotrophic pathogens have adopted strategies to induce senescence by manipulating the respective signaling pathways in plants. Consequently, a defense strategy against necrotrophs would consist of the host’s ability to prevent senescence-like processes. Conversely, biotrophic pathogens benefit from living cells and change signaling processes in the host to delay senescence. The most efficient defense strategy against biotrophs is therefore the controlled induction of cell death during the hypersensitive response (HR), thus starving the pathogen in confined regions. It can easily be imagined that the situation is especially complex for hemibiotrophs. Resistance against necrotrophs and hemibiotrophs is often quantitative, i.e., restricting, but not eliminating the pathogen \cite{plants4030449}.

We have used a data integrative approach to screen a large population of wheat plants to try to understand the interaction plant - pathogens and its influence on the onset of senescence both at the leaf and whole plant level. 


%\begin{methods}
\section{Material \& Methods}

\subsection{Plant Material}
The eight-parent MAGIC population described by \cite{Mackay01092014} was used for all phenotypic screening. Briefly, the complete population consists of more than >1000 lines generated from three rounds of inter-crossing between eight elite United Kingdom wheat varieties (Alchemy, Brompton, Claire, Hereward, Rialto, Robigus, Soissons, Xi19) followed by five rounds of selfing to remove heterozygosity. 

\subsection{Glass House Cultivation}
An experiment  was conducted between mid January 2015 and mid April 2015 in The National Plant Phenomics Centre greenhouse facilities in Aberystwyth, UK. The MAGIC parents as well as 208 lines (see Supporting Information, Table S1) were grown under a well watered conditions with two replicates per genotype. 

Single plants were grown in 15cm sq plastic pots. A number of seeds per pot were directly sown into the soil and after germination thinned out, leaving one plant per pot. Plants were put in vernalisation for nine weeks and later taken to regular greenhouse. Watering was performed manually to allow optimal germination and seedling establishment. Subsequently, the pots were transferred to the “smart house” where each pot was placed onto a cart on a conveyor belt. Every day, pots were weighed and watered automatically to 75\% gravimetric water content (Fig. 2). The experiments were carried out under natural lighting with the temperature in the greenhouse kept at a range between 15°C (night) and 18°C (day).

\subsection{Plant infection}
Wheat plants were naturally infected with tan spot which is caused by the fungus Pyrenophora tritici-repentis. Pyrenophora tritici-repentis is a necrotrophic plant pathogen of fungal origin Phylum Ascomycota. Tan spot symptoms can be manifested as necrosis, chlorosis or both. The necrosis symptom comprises spots that appear initially as tan-brown flecks and expand into lens-shaped, tan lesions with yellow borders. The chlorosis symptom consists of rapidly expanding yellow areas surrounding lesions on the leaf blades. Lesions may coalesce into large blotches as they age, predisposing leaves to premature senescence.  If infected leaves are moistened, the lesions darken at the center due to formation of conidiophores and conidia of D. tritici-repentis \cite{Singh2009}. 
\subsection{Disease scoring}
Tan spot infection was scored manually using the seedling infection type (IT) score shown in Table \ref{Tab:02}. These qualitative IT scores were converted to a numerical scale for all statistical analysis.

\subsection{Phenotyping}
Plants were imaged every day, from the time they were put on the belts. The LemnaTec 3D Scanalyzer (LemnaTec, GmbH, Wuerselen, Germany) for image acquisition. Three RGB pictures (2056×2454 pixels) were taken of each wheat plant, one top view image and two side view images with a 90° horizontal rotation. Plants were also manually scored for developmental stages using the AHDB growth guide. A glass house operator monitored the plants twice a week and recorded the dates when evey single plants reached gs39, gs55, gs65 and flag leaf senescence developmental stages. At the end of the experiment, wheat plants were harvested and above ground biomass, tiller number (\textit{TN}), plant height (PH), steam height (SH), top internode length (TIL), first/second/third ear length (1EL,2EL, 3EL), first ear weight (1EW), flag leaf senescence (FLS) and first flag leaf length (1FLL) were determined.

% Fresh biomass was weighed and, subsequently, oven dried to constant weight to determine dry biomass. Water use efficiency (WUE) was calculated by dividing dry biomass at the end of the experiment by the total amount of water added during the four weeks in the “smart house” [mg/g water]. Specific plant weight (SPW) was calculated from the dry weight and the maximum projected shoot area at the end of the experiment. 
\subsection{Image processing}
Digital images were processed using the \cite{MATLAB:2010}
After background-foreground separation was applied to separate the plant tissue area from the background, pixel numbers per plant were counted and the pixel sum of the three pictures per plant was taken to define the biomass %{projected shoot area. The shoot area measured over time was used to draw growth curves. For each growth curve, curve fitting with a 6th order polynomial was conducted to adjust for possible missing data points and absolute growth rate [dA/dt] and relative growth rate [(dA/dt)/A] were calculated. For each of the three curves the integral was determined and used as a trait in the statistical analysis}. 


\begin{table}[!t]
\textbf{\refstepcounter{table}\label{Tab:02} Table \arabic{table}.}{ Infection type score table}\\
\processtable{}
{\begin{tabular}{l|l}
\hline
	score & description \\\midrule
	0 & No visible symptoms \\
	1  & Small, sporulating uredia surrounded by necrotic tissue \\
	2 & Small, sporulating uredia surrounded by necrotic tissue \\
	3 & Medium sized, sporulating uredia surrounded only by chlorotic tissue \\
	4 & Large, sporulating uredia surrounded by green tissue \\ \hline
\end{tabular}}{}
\end{table}


\subsection{Statistics and QTL analysis}
Statistical analyses were performed using the R environment \cite{RManual}. The analysis of multi-parental populations requires a different approach, because single marker association or interval mapping can fail to detect a QTL if the causative alleles do not segregate between the founders in the same way as the individual markers \cite{Mott07112000}. Furthermore, when mapping QTL in structured populations the evidence for the existence of a QTL has to be considered in the context of other QTL which might explain some of the same component of variation. Population structure can produce long-range correlations between genotypes and hence “ghost” QTL, although the LD analysis suggests that the MAGIC population is relatively immune to this phenomenon. To deal with these issues and model the mosaic structure we used the the HAPPY package from the R environment \cite{Mott07112000}. More details about the use HAPPY with other MAGIC population are in \cite{10.1371/journal.pgen.1000551}

\section{Results}
Tan spot resistance was evaluated in the greenhouse by measuring the extent of the necrotic/chlorotic response exhibited by the plant (IT). To avoid bias in the scoring, two people scored the plants at two time points and scores were averaged. When comparing disease scores against the onset of senescence at the Flag Leaf (FLS), we found that more severly infected plants started senescence earlier that those mildy infected Fig \ref{fig2}. This result is consistent with the way necrotrophic plant pathogens act, they derive nutrients from dead cells. Therefore, promoting senescence in the host by manipulating respective signaling pathways is a mechanism they use to speed up cell death.  

Looking back into the development of the plant prior to senescence, we realised there was also an asociation between the speed at which plants developed and disease severity. Supplementary Figures 1-3 shows plots mapping the date (DAS) at which plants reached GS39, GS55 and GS65 plant developmental stages against their corresponding disease severity score. The trend lines in the plots suggest there was an association between disease severity and the speed at which plants developed. This result is also consistent with the theory that necrotrophic plant pathogens act, they derive nutrients from dead cells. Therefore, promoting senescence in the host by manipulating respective signaling pathways is a mechanism they use to speed up cell death. It's not surprised we found high positive correlations between all three developmental stages and the onset of senescence at the flag leaf (Supplementary Figure 4).

\subsection{MLs}

As well as looking at the whole population, we also identified several RILs who managed to delay senescence despite been severely affected (score four) by the pathogen. Fig \ref{fig3} shows a contrast between two plants highly susceptible to pathogen infection but whose initiation of senescence ocurred at two different times. Fig \ref{fig3}b is a plant severely infected by the pathogen but who manages to delay the start of senescence at the flag leaf. 

\subsection{Quantitative trait loci}

QTL analysis identified the genomic regions associated to the onset of gs39 plant developmental stage and for the traits first.ear.weight and third.ear.length. For the gs39 trait, the analysis identified three QTLs on chromosome 5A between ?? and ?? Mb (90\% confidence region), EQ =??, peak logP =10.38, 4.34 and 5.09 \ref{fig:gs39}. Another QTL was found on chromosome 5D between ?? and ?? Mb (90\% confidence region), EQ = ??, peak logP =4.16. For the first.ear.weight,  the analysis identified a single QTL on chromosome 1A between ?? and ?? Mb (90\% confidence region), EQ =??, peak logP = 6.24 \ref{fig:firstearweight}. For the third.ear.length,  the analysis identified a single QTL on chromosome 5B between ?? and ?? Mb (90\% confidence region), EQ =??, peak logP = 4.00 \ref{fig:thirdearlength}. Table \ref{table:qtl} shows a QTL summary for each trait. The rest of the traits including disease score show no marker associations. 
\todo{I need to add more infor here. What is the relevance of this QTLs if any?}

\begin{table}[ht]
 \textbf{\refstepcounter{table}\label{table:qtl} Table \arabic{table}.}{ logP is -log$_{10}$ (ANOVA P-value) at the QTL peak; chr is the chromosome}\\
\input{qtltable.tex}
\end{table}


\section{Discussion}

The study has used a data integrative approach to 
\section*{Disclosure/Conflict-of-Interest Statement}


The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.

\section*{Author Contributions}

Conceived and designed the study: AVC/AB. Performed the study: AVC. Analyzed the data: AVC JTK. Contributed with seed: AB. Wrote the paper: AVC JTK. Provided comments and corrected the manuscript: ALL.

\section*{Acknowledgments}
 We are grateful to the team of “National Plant Phenomics Centre” for carrying out the experiments and to the 
\textit{Funding\textcolon} for finantially suportting the project.



\bibliographystyle{frontiersinSCNS_ENG_HUMS} % for Science, Engineering and Humanities and Social Sciences articles, for Humanities and Social Sciences articles please include page numbers in the in-text citations
%\bibliographystyle{frontiersinHLTH&FPHY} % for Health and Physics articles
\bibliography{bioinfo}

%%% Upload the *bib file along with the *tex file and PDF on submission if the bibliography is not in the main *tex file

\section*{Figures}

%%% Use this if adding the figures directly in the mansucript, if so, please remember to also upload the files when submitting your article
%%% There is no need for adding the file termination, as long as you indicate where the file is saved. In the examples below the files (logo1.jpg and logo2.eps) are in the Frontiers LaTeX folder
%%% If using *.tif files convert them to .jpg or .png

\begin{figure}
    \begin{center}
    \begin{subfigure}[b]{0.5\textwidth}
        \includegraphics[width=\textwidth]{smarthouse.jpg}
        \caption{Plant watering at gravimetrix level}
       % \label{fig:tiger}
    \end{subfigure}
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
    %(or a blank line to force the subfigure onto a new line)
    \begin{subfigure}[b]{0.5\textwidth}
        \includegraphics[width=\textwidth]{scoring.jpg}
        \caption{Plant scoring for developmental stages}
        %\label{fig:mouse}
    \end{subfigure}
\end{center}
    \caption{Contrast beween two lines whose disease score was four. Red vertical line indicates the date senescence on Flag leaf was first seen}\label{fig3}
%\textbf{\refstepcounter{figure}\label{fig:03} Figure \arabic{figure}.}{Contrast beween two lines whose disease score was four. Red vertical line indicates the date senescence on Flag leaf was first seen}
\end{figure}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=10cm]{flowchart.jpg}
\end{center}
 \caption{ Whole leaf senescence flowchart}\label{fig1:flowchart}
\end{figure}

\begin{figure}[h!]
\begin{center}
\includegraphics[width=10cm]{flagleafsenescence.png}
\end{center}
 \caption{Disease severity vs onset of senescence at the Flag Leaf. Red dots indicate average DAS per disease score}\label{fig2}
\end{figure}


\begin{figure}
    \begin{center}
    \begin{subfigure}[b]{0.5\textwidth}
        \includegraphics[width=\textwidth]{W8-214112.png}
        \caption{Early senescence, IT = 4}
       % \label{fig:tiger}
    \end{subfigure}
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
    %(or a blank line to force the subfigure onto a new line)
    \begin{subfigure}[b]{0.5\textwidth}
        \includegraphics[width=\textwidth]{W8-100112.png}
        \caption{Late senescence, IT = 4}
        %\label{fig:mouse}
    \end{subfigure}
\end{center}
    \caption{Contrast beween two lines whose disease score was four. Red vertical line indicates the date senescence on Flag leaf was first seen}\label{fig3}
%\textbf{\refstepcounter{figure}\label{fig:03} Figure \arabic{figure}.}{Contrast beween two lines whose disease score was four. Red vertical line indicates the date senescence on Flag leaf was first seen}
\end{figure}


\begin{figure}
    \begin{center}
    \begin{subfigure}[b]{1\textwidth}
        \includegraphics[width=\textwidth]{gs39_qtl.png}
        \caption{}
        \label{fig:gs39}
    \end{subfigure}
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
    %(or a blank line to force the subfigure onto a new line)
\begin{subfigure}[b]{1\textwidth}
        \includegraphics[width=\textwidth]{firstearweight_qtl.png}
        \caption{}
       \label{fig:firstearweight}
    \end{subfigure}
    ~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
    %(or a blank line to force the subfigure onto a new line)
    \begin{subfigure}[b]{1\textwidth}
        \includegraphics[width=\textwidth]{thirdearlength_qtl.png}
        \caption{}
        \label{fig:thirdearlength}
    \end{subfigure}
\end{center}
    \caption{QTLs hits for the taits gs39, first.ear.weightt and third.ear.length}\label{fig4}
%\textbf{\refstepcounter{figure}\label{fig:03} Figure \arabic{figure}.}{Contrast beween two lines whose disease score was four. Red vertical line indicates the date senescence on Flag leaf was first seen}


\end{figure}


%\begin{figure}
%\begin{center}
%\includegraphics[width=10cm]{logo2}% This is an *.eps file
%\end{center}
%\textbf{\refstepcounter{figure}\label{fig:02} Figure \arabic{figure}.}{ Enter the caption for your figure here.  Repeat as  necessary for each of your figures }
%\end{figure}

%%% If you don't add the figures in the LaTeX files, please upload them when submitting the article.

%%% Frontiers will add the figures at the end of the provisional pdf automatically %%%

%%% The use of LaTeX coding to draw Diagrams/Figures/Structures should be avoided. They should be external callouts including graphics.
\end{document}
